e2e/analysis/analysis_test.go

```diff
@@ -0,0 +1,138 @@
+	"github.com/ZupIT/horusec-devkit/pkg/enums/tools"
+	"github.com/ZupIT/horusec/internal/utils/testutil"
+)
+
+type AnalysisTestCase struct {
+	Language       languages.Language
+	AnalysisFolder string
+	ExpectedTools  []tools.Tool
+}
+
+func (A AnalysisTestCase) RunAnalysisTestCase() (*gexec.Session, error) {
+	flags := map[string]string{
+		testutil.StartFlagProjectPath: A.AnalysisFolder,
+	}
+	cmd := testutil.GinkgoGetHorusecCmdWithFlags(testutil.CmdStart, flags)
+	return gexec.Start(cmd, GinkgoWriter, GinkgoWriter)
+}
+
+var testcases = []AnalysisTestCase{
+	{
+		Language:       languages.Python,
+		AnalysisFolder: testutil.PythonExample1,
+		ExpectedTools:  []tools.Tool{tools.Bandit, tools.BundlerAudit, tools.Trivy},
+	},
+	{
+		Language:       languages.Ruby,
+		AnalysisFolder: testutil.RubyExample1,
+		ExpectedTools:  []tools.Tool{tools.Brakeman, tools.BundlerAudit, tools.Trivy},
+	},
+	{
+		Language:       languages.Javascript,
+		AnalysisFolder: testutil.JavaScriptExample1,
+		ExpectedTools:  []tools.Tool{tools.NpmAudit, tools.YarnAudit, tools.HorusecEngine, tools.Semgrep, tools.Trivy},
+	},
+	{
+		Language:       languages.Go,
+		AnalysisFolder: testutil.GoExample1,
+		ExpectedTools:  []tools.Tool{tools.GoSec, tools.Semgrep, tools.Nancy, tools.Trivy},
+	},
+	{
+		Language:       languages.CSharp,
+		AnalysisFolder: testutil.CsharpExample1,
+		ExpectedTools:  []tools.Tool{tools.SecurityCodeScan, tools.HorusecEngine, tools.DotnetCli, tools.Trivy},
+	},
+	{
+		Language:       languages.Java,
+		AnalysisFolder: testutil.JavaExample1,
+		ExpectedTools:  []tools.Tool{tools.HorusecEngine, tools.Semgrep, tools.Trivy},
+	},
+	{
+		Language:       languages.Kotlin,
+		AnalysisFolder: testutil.KotlinExample1,
+		ExpectedTools:  []tools.Tool{tools.HorusecEngine},
+	},
+	{
+		Language:       languages.Leaks,
+		AnalysisFolder: testutil.LeaksExample1,
+		ExpectedTools:  []tools.Tool{tools.GitLeaks, tools.HorusecEngine},
+	},
+	{
+		Language:       languages.PHP,
+		AnalysisFolder: testutil.PHPExample1,
+		ExpectedTools:  []tools.Tool{tools.Semgrep, tools.PhpCS, tools.Trivy},
+	},
+	{
+		Language:       languages.Dart,
+		AnalysisFolder: testutil.DartExample1,
+		ExpectedTools:  []tools.Tool{tools.HorusecEngine},
+	},
+	{
+		Language:       languages.Elixir,
+		AnalysisFolder: testutil.ElixirExample1,
+		ExpectedTools:  []tools.Tool{tools.MixAudit, tools.Sobelow},
+	},
+	{
+		Language:       languages.Nginx,
+		AnalysisFolder: testutil.NginxExample1,
+		ExpectedTools:  []tools.Tool{tools.HorusecEngine},
+	},
+	{
+		Language:       languages.Swift,
+		AnalysisFolder: testutil.SwiftExample1,
+		ExpectedTools:  []tools.Tool{tools.HorusecEngine},
+	},
+}
+
+var _ = Describe("Run a complete horusec analysis", func() {
+	var (
+		session *gexec.Session
+	)
+
+	for _, tt := range testcases {
+		Describe(fmt.Sprintf("Running on %s codebase.", tt.Language.ToString()), func() {
+			session, _ = tt.RunAnalysisTestCase()
+			session.Wait(2 * time.Minute)
```

Comment:
We could use the docker event [monitor](https://pkg.go.dev/github.com/docker/docker/client#Client.Events) here instead of waiting some time, i think it will reduce quite some time on this e2e tests

Reply:
This time works as a timeout, not a "sleep".